src/System.Binary/System/Binary/BufferReader.cs

```diff
@@ -105,5 +105,33 @@ public static void RequiresInInclusiveRange(int start, uint length)
                 throw new ArgumentOutOfRangeException();
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe static int ReverseEndianness(this int num)
+        {
+            uint val = 0;
```

Comment:
BTW, now that Reverse methodfs are non-generic, you don't need to Read and Write.